agent/src/main/scala/za/co/absa/atum/agent/AtumAgent.scala

```diff
@@ -18,55 +18,49 @@ package za.co.absa.atum.agent
 
 import com.typesafe.config.{Config, ConfigFactory}
 import za.co.absa.atum.agent.AtumContext.AtumPartitions
-import za.co.absa.atum.agent.dispatcher.{ConsoleDispatcher, HttpDispatcher}
+import za.co.absa.atum.agent.dispatcher.{CapturingDispatcher, ConsoleDispatcher, Dispatcher, HttpDispatcher}
 import za.co.absa.atum.model.dto.{AdditionalDataSubmitDTO, CheckpointDTO, PartitioningSubmitDTO}
 
 /**
- * Entity that communicate with the API, primarily focused on spawning Atum Context(s).
+ *  Entity that communicate with the API, primarily focused on spawning Atum Context(s).
  */
-class AtumAgent private[agent] () {
+abstract class AtumAgent private[agent] () {
```

Comment:
```suggestion
abstract class AtumAgent private[agent] {
```

Reply:
I actually wonder. why it's an (abstract) class and not trait and have private constructor 🤔

Reply:
Yeah I think that it's an evolution and currently it's just too 'complicated'. I can't came up with any good reasons why it shouldn't be a trait. Let's change it :)